client/state/notices/actions.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Internal dependencies
+ */
+import {
+	NEW_NOTICE,
+	REMOVE_NOTICE
+} from 'state/action-types';
+
+import { uniqueId } from 'lodash';
```

Comment:
That is correct, they will be unique across.
Memory wont probably be a problem, because max integer is `9007199254740991` so that's a lot of notices and I suspect we will run into many other leaks before this becomes an issue.
That being said, the problem here is state (de)hydration

Let's say we are dehydrating notices and next id is 4.
When we reload and  re-hydrate, the next id is 1 and we already have notice with id 1 in store.

So, I intend to make a uniqueID in global store with state saving
But this is  probably a matter for another PR :)
Probably can started on that right away.